core/src/main/scala/design/hamu/Game.scala

```diff
@@ -0,0 +1,21 @@
+package design.hamu
+
+class Game(var board: Board, var isActive: Boolean, var turn: Char) {
+
+  def convertToCoordinates(inputPosition: String): (Int, Int) = {
+    (inputPosition.toInt % 3, (inputPosition.toInt - inputPosition.toInt % 3)/3)
+  }
+
+  def isMoveValid (inputPosition: String) : Boolean = {
+    val (x, y): (Int, Int) = convertToCoordinates(inputPosition)
+    board.get(x, y) == '_'
+  }
+
+  def input(inputPosition: String) : Board = {
```

Comment:
No big deal but you could have this return a new Game like so:
```
def input(inputPosition: String): Board = {
   val (x, y) = convertToCoordinates(inputPosition)
   val b2 = board.move(x, y, turn)
   new Game(b2, checkFunction(board), if(turn == 'O') 'X' else 'O')
}
```
and then you don't have to wire stuff outside